File: Bitwise/Medium/LET_no1239_Maximum Length of a Concatenated String with Unique Characters.cpp
class Solution {
public:
    int maxLength(vector<string>& arr) {
        vector<bitset<26>> dp = {bitset<26>()};
        int res = 0;
        for (auto& s : arr) {
            bitset<26> a;
            for (char c : s)
                a.set(c - 'a');
            int n = a.count();
            if (n < s.size())
                continue;
            
            for (int i = dp.size() - 1; i >= 0; i--) {
                bitset b = dp[i];
                if ((a & b).any())
                    continue;
                dp.push_back(a | b);
                res = max(res, (int) b.count() + n);
            }
        }
        return res;
    }
};

File: Data Structure/Hash/Medium/LET_no1239_Maximum Length of a Concatenated String with Unique Characters.py
class Solution:
    def maxLength(self, arr: List[str]) -> int:
        dp = [set()]
        for a in arr:
            if len(set(a)) < len(a):
                continue
            a = set(a)
            for b in dp[:]:
                if a.intersection(b):
                    continue
                dp.append(a.union(b))
        return max(len(c) for c in dp)